timers/timers_test.go

```diff
@@ -0,0 +1,417 @@
+package timers
+
+import (
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/andres-erbsen/clock"
+	"github.com/stretchr/testify/assert"
+)
+
+// O(n), only for use in tests.
+func (t *Timer) len() int {
+	n := 0
+	for el := t; el != nil; el = el.next {
+		n++
+	}
+	return n
+}
+
+func fakeWheel(tick time.Duration) (*Wheel, *clock.Mock) {
+	clock := clock.NewMock()
+	return newWheel(tick, 2*time.Minute, clock), clock
+}
+
+func newBucketList() *bucketList {
+	return &bucketList{
+		mask:    uint64(1<<3 - 1),
+		buckets: make([]bucket, 1<<3),
+	}
+}
+
+func assertQueuedTimers(t testing.TB, bs *bucketList, bucketIdx, expected int) {
+	b := &bs.buckets[bucketIdx]
+	b.Lock()
+	defer b.Unlock()
+	assert.Equal(
+		t,
+		expected,
+		b.timers.len(),
+		"Unexpected number of counters in bucket %v.", bucketIdx,
+	)
+}
+
+func fakeWork() {} // sentinel non-nil func to schedule
```

Comment:
i don't think this is a [sentinel value](https://en.wikipedia.org/wiki/Sentinel_value), just a noop func.

sentintel implies that something else is looking for this specific value, but this is just a normal function as far as the timer wheel is concerned